build/scripts/GitHub/GitHubPullRequest.class.psm1

```diff
@@ -0,0 +1,68 @@
+using module .\GitHubAPI.class.psm1
+using module .\GitHubIssue.class.psm1
+
+
+<#
+    Class that represents a GitHub pull request.
+#>
+class GitHubPullRequest {
+    [int] $PRNumber
+    [string] $Repository
+    $PullRequest
+
+    hidden GitHubPullRequest([int] $PRNumber, [string] $Repository) {
+        $this.PRNumber = $PRNumber
+        $this.Repository = $Repository
+
+        $pr = gh api "/repos/$Repository/pulls/$PRNumber" -H ([GitHubAPI]::AcceptJsonHeader) -H ([GitHubAPI]::GitHubAPIHeader) | ConvertFrom-Json
+        if ($pr.message) {
+            # message property is populated when the PR is not found
+            throw "::Error:: Could not get PR $PRNumber from repository $Repository. Error: $($pr.message)"
+        }
+
+        $this.PullRequest = $pr
+    }
+
+    <#
+        Gets the pull request from GitHub.
+    #>
+    static [GitHubPullRequest] Get([int] $PRNumber, [string] $Repository) {
+        $pr = [GitHubPullRequest]::new($PRNumber, $Repository)
+
+        return $pr
+    }
+
+    [string] GetBody() {
+        return $this.PullRequest.body
+    }
+
+    <#
+        Removes a comment from the pull request if it exists.
+    #>
+    RemoveComment($Message) {
+        $existingComments = gh api "/repos/$($this.Repository)/issues/$($this.PRNumber)/comments" -H $this.AcceptJsonHeader -H $this.GitHubAPIHeader  | ConvertFrom-Json
```

Comment:
Since we're calling the issues api, wouldn't it make sense to place the "comment" functions in the GitHubIssue class? 
Maybe the RemoveComment method in this class can just create a GitHubIssue object and call RemoveComment on that?

Reply:
It's the issue endpoint because issues and PRs on GitHub share a lot of functionality, e.g. the comment section.

In this case I have a strong opinion that this should stay in the PR class.